tsdate/core.py

```diff
@@ -957,47 +939,104 @@ def __init__(
                 population_size=Ne.as_dict() if hasattr(Ne, "as_dict") else Ne,
             )
 
-        if priors is None:
-            if Ne is None:
+        if constr_iterations is None:
+            self.constr_iterations = 0
+        else:
+            if not (isinstance(constr_iterations, int) and constr_iterations >= 0):
                 raise ValueError(
-                    "Must specify population size if priors are not already built using"
-                    f"tsdate.build_{self.prior_grid_func_name}()"
+                    "Number of constrained least squares iterations must be a "
+                    "non-negative integer"
                 )
-            mk_prior = getattr(prior, self.prior_grid_func_name)
-            # Default to not creating approximate priors unless ts has
-            # greater than DEFAULT_APPROX_PRIOR_SIZE samples
-            approx = True if ts.num_samples > base.DEFAULT_APPROX_PRIOR_SIZE else False
-            self.priors = mk_prior(ts, Ne, approximate_priors=approx, progress=progress)
-        else:
-            logging.info("Using user-specified priors")
+            self.constr_iterations = constr_iterations
+
+        if self.prior_grid_func_name is None:
+            if priors is not None:
+                raise ValueError(f"Priors are not used for method {self.name}")
             if Ne is not None:
-                raise ValueError(
-                    "Cannot specify population size if specifying priors "
-                    f"from tsdate.build_{self.prior_grid_func_name}()"
+                raise ValueError(f"Population size is not used for method {self.name}")
```

Comment:
For future ref, see https://github.com/tskit-dev/tsdate/issues/376 which will remove the need to do this.